apps/framework-cli/src/cli/routines.rs

```diff
@@ -266,6 +267,29 @@ impl RoutineController {
     }
 }
 
+async fn leadership_callback() -> Result<(), anyhow::Error> {
+    info!("This instance is the leader. Starting cron registry");
+    let cron_registry: CronRegistry = CronRegistry::new()
+        .await
+        .map_err(|e| anyhow::anyhow!(Box::<dyn std::error::Error + Send + Sync>::from(e)))?;
+    cron_registry.load_jobs().await?;
+    cron_registry.start().await?;
+    Ok(())
+}
+
+async fn setup_redis_client(project: &Project) -> anyhow::Result<RedisClient> {
+    let mut redis_client = RedisClient::new(project.name(), project.redis_config.clone()).await?;
+    redis_client.set_leadership_callback(|| {
+        tokio::spawn(async {
```

Comment:
It's an effort to avoid an HTTP request that blocks - this also allows multiple jobs that might execute on the same time spec to not block each other.